app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt

```diff
@@ -165,27 +167,43 @@ class UseCaseModule {
     @ViewModelScoped
     @Provides
     fun listenToEventsUseCaseProvider(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentSession session: AuthSession) =
-        coreLogic.getSessionScope(session).listenToEvents
+        coreLogic.getSessionScope(session.userId).listenToEvents
 
     @ViewModelScoped
     @Provides
     fun providesGetSelfUseCase(
         @KaliumCoreLogic coreLogic: CoreLogic,
         @CurrentSession currentSession: AuthSession
     ): GetSelfUserUseCase =
-        coreLogic.getSessionScope(currentSession).users.getSelfUser
+        coreLogic.getSessionScope(currentSession.userId).users.getSelfUser
 
     @ViewModelScoped
     @Provides
     fun providesSendTextMessageUseCase(
         @KaliumCoreLogic coreLogic: CoreLogic,
         @CurrentSession currentSession: AuthSession
     ): SendTextMessageUseCase =
-        coreLogic.getSessionScope(currentSession).messages.sendTextMessage
+        coreLogic.getSessionScope(currentSession.userId).messages.sendTextMessage
+
+    @ViewModelScoped
+    @Provides
+    fun providesSearchKnownUsersUseCase(
```

Comment:
What do you think of calling knownUsers as connectedUsers?

Reply:
do you mean changing the name only on the reloaded side or also Kalium ?

Reply:
I would say both, as we should keep it as consistent as possible between client and core

Reply:
there was lots of disscussion ongoing about the naming and this is something we agreed so far on Kalium side, so I am going to keep it for now